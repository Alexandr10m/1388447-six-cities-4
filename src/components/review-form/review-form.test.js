import React from "react";
import renderer from "react-test-renderer";
import ReviewForm from "./review-form.jsx";


describe(`Snapshot of ReviewForm`, () => {
  it(`Should render correctly`, () => {
    const tree = renderer
      .create(
          <ReviewForm
            commentText={`hello`}
            buttonDisable={false}
            isValidText={false}
            isValidRate={false}
            disabledForm={false}
            onSubmit={()=>{}}
            onRatingChange={()=>{}}
            offerId={1}
            onUserCommentEnter={()=>{}}
          >
            <div/>
          </ReviewForm>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
