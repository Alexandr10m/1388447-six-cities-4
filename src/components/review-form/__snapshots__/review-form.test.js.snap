// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of ReviewForm Should render correctly 1`] = `
<form
  action="#"
  className="reviews__form form"
  method="post"
  onSubmit={[Function]}
>
  <label
    className="reviews__label form__label"
    htmlFor="review"
  >
    Your review
  </label>
  <div
    className="reviews__rating-form form__rating"
    onChange={[Function]}
  >
    <div />
  </div>
  <textarea
    className="reviews__textarea form__textarea"
    disabled={false}
    id="review"
    name="review"
    onChange={[Function]}
    placeholder="Tell how was your stay, what you like and what can be improved"
    value="hello"
  />
  <div
    className="reviews__button-wrapper"
  >
    <p
      className="reviews__help"
    >
      To submit review please make sure to set 
      <span
        className="reviews__star"
      >
        rating
      </span>
       and describe your stay with at least 
      <b
        className="reviews__text-amount"
      >
        50 characters
      </b>
      .
    </p>
    <button
      className="reviews__submit form__submit button"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
  <div
    style={
      Object {
        "color": "red",
      }
    }
  >
     You need to rate
  </div>
  <div
    style={
      Object {
        "color": "red",
      }
    }
  >
     You need to enter 50 - 300 characters, but entered 
    5
  </div>
</form>
`;
